src/util/simplify_expr.cpp

```diff
@@ -1734,12 +1734,15 @@ bool simplify_exprt::simplify_byte_extract(byte_extract_exprt &expr)
     return true;
 
   if(expr.op().id()==ID_array_of &&
-     expr.op().op0().id()==ID_constant)
+     to_array_of_expr(expr.op()).op().id()==ID_constant)
   {
     std::string const_bits=
-      expr2bits(expr.op().op0(),
+      expr2bits(to_array_of_expr(expr.op()).op(),
                 byte_extract_id()==ID_byte_extract_little_endian);
 
+    if(const_bits.empty())
```

Comment:
Add a comment suggesting why we would have a zero-width constant like this?

Reply:
I'm tempted to say that this should be an invariant? (It obviously isn't as that unbounded recursion has been observed, but I'm equally surprised by that.)

Reply:
I think it could return an optional, which, when without value, would signal that there is no bit representation.
Then !const_bits.empty() would be an invariant.